compiler/nim.nim

```diff
@@ -37,7 +37,9 @@ proc prependCurDir(f: string): string =
   else:
     result = f
 
-proc processCmdLine(pass: TCmdLinePass, cmd: string; config: ConfigRef) =
+type ProgNim=ref object of ProgBase
```

Comment:
Empty subclass means you don't need any inheritance/OOP here.

Reply:
done, changed it to not use OOP